framework/xds_k8s_testcase.py

```diff
@@ -1382,3 +1388,99 @@ def debug_cert(cert):
+            project=self.project,
+            resource_prefix=self.resource_prefix,
+            resource_suffix=self.resource_suffix,
+            network=self.network,
+            compute_api_version=self.compute_api_version,
+            enable_dualstack=self.enable_dualstack,
+        )
+
+    def initKubernetesClientRunner(self, **kwargs) -> KubernetesClientRunner:
+        return KubernetesClientRunner(
+            k8s.KubernetesNamespace(
+                self.k8s_api_manager, self.client_namespace
+            ),
+            deployment_name=self.client_name,
+            deployment_template="client.cloudrun.deployment.yaml",
+            image_name=self.cloudrun_client_image,
+            td_bootstrap_image=self.td_bootstrap_image,
+            gcp_project=self.project,
+            gcp_api_manager=self.gcp_api_manager,
+            gcp_service_account=self.gcp_service_account,
+            xds_server_uri=self.xds_server_uri,
+            network=self.network,
+            debug_use_port_forwarding=self.debug_use_port_forwarding,
+            enable_workload_identity=self.enable_workload_identity,
+            stats_port=self.client_port,
+            reuse_namespace=self.server_namespace == self.client_namespace,
+            **kwargs,
+        )
+
+    def setupServerBackends(
+        self,
+        *,
+        wait_for_healthy_status=True,
+        server_runner=None,
+        max_rate_per_endpoint: Optional[int] = None,
+    ):
+        if server_runner is None:
+            server_runner = self.server_runner
+        service_url = server_runner.get_service_url()
+        self.td.backend_service_add_backends([service_url])
+
+    def startTestServers(
+        self, server_runner=None, **kwargs
+    ) -> List[XdsTestServer]:
+        logger.info(self.cloudrun_server_image)
+        if server_runner is None:
+            self.server_runner = CloudRunServerRunner(
+                project=self.project,
+                service_name=self.server_namespace,
+                image_name=self.cloudrun_server_image,
+                network=self.network,
+                region=self.region,
+            )
+        test_servers = self.server_runner.run()
+        for test_server in test_servers:
+            test_server.set_xds_address(
+                self.server_xds_host, self.server_xds_port
+            )
+        return test_servers
+
+    def startTestClient(
+        self, test_server: XdsTestServer, **kwargs
+    ) -> XdsTestClient:
+        return self._start_test_client(test_server.xds_uri, **kwargs)
+
+    def backend_service_add_serverless_neg_backends(self):
+        logger.info("Creating serverless NEG")
+        neg = self.compute_v1.create_serverless_neg(
+            self.server_namespace,
+            self.region,
+            self.server_namespace,
+            "serverless",
+        )
+        return neg
+
+    def tearDown(self):
+        logger.info("----- TestMethod %s teardown -----", self.test_name)
```

Comment:
add TODO to check for client crashes similar to `IsolatedXdsKubernetesTestCase.tearDown`